lib/listen/fsm.rb

```diff
@@ -1,131 +1,128 @@
 # Code copied from https://github.com/celluloid/celluloid-fsm
+
+require 'thread'
+
 module Listen
   module FSM
-    DEFAULT_STATE = :default # Default state name unless one is explicitly set
+    START_STATE = :default # Start state name unless one is explicitly set
 
     # Included hook to extend class methods
     def self.included(klass)
       klass.send :extend, ClassMethods
     end
 
     module ClassMethods
-      # Obtain or set the default state
-      # Passing a state name sets the default state
-      def default_state(new_default = nil)
-        if new_default
-          @default_state = new_default.to_sym
+      # Obtain or set the start state
+      # Passing a state name sets the start state
+      def start_state(new_start_state = nil)
+        if new_start_state
+          new_start_state.is_a?(Symbol) or raise ArgumentError, "state name must be a Symbol (got #{new_start_state.inspect})"
```

Comment:
Instead of explicitly detecting it's a symbol, why not use `to_sym`?

Reply:
This kind of pseudo type checking is uncommon in Ruby, even thought I can see how it makes sense.

The reality is, if you add this, there are many other places where you might start trying to enforce type contracts.

Agreed, we don't really want users passing in strings.

Reply:
True that it's not common in Ruby (yet). But we've found it to be invaluable for containing String vs. Symbol insanity! Otherwise, you wind up with defensive code sprinkled everywhere that tries to handle String and Symbol as equivalent...and you never know if that clutter is complete.

Speaking of which, I added the missing spec for the FSM module and made sure to test this case: https://github.com/guard/listen/pull/482/commits/bbd10eecca86cfdae4f73a3a4906d8554f086b19